galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunResultRas.java

```diff
@@ -28,27 +30,28 @@
 
     @Reference
     IFramework framework;
+    
+    private final Gson gson = GalasaGsonBuilder.build();
 
     private static final long serialVersionUID = 1L;
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
        
        
-        Gson gson = GalasaGsonBuilder.build();
-       
         String runId = "";
         String url = req.getRequestURI().toString();
-        String[] parts = url.split("/");
         
+        Pattern pattern = Pattern.compile("(?!.*\\/).+");
+        Matcher matcher = pattern.matcher(url);
         
         String json = "";
        
          try {
             
             //Check if there is an id
-            if (parts[3] != null) {
-               runId = parts[3];
+            if (matcher.find()) {
```

Comment:
what happens if it doesn't match?   you need to return an something invalid response code